packages/itwin/grouping-mapping-widget/src/components/Mappings/hooks/useMappingsOperations.ts

```diff
@@ -33,29 +36,25 @@ const fetchExtractionStatus = async (
   iModelId: string,
   getAccessToken: GetAccessTokenFn,
   extractionClient: IExtractionClient,
-  setExtractionIconData: React.Dispatch<React.SetStateAction<ExtractionIconData>>,
-  setExtractionMessageData: React.Dispatch<React.SetStateAction<ExtractionMessageData[]>>
+  setExtractionStatusIcon: (extractionStatusIcon: ExtractionStatusData | ((extractionStatusIcon: ExtractionStatusData) => ExtractionStatusData)) => void,
+  setExtractionMessageData: (extractionMessageData: ExtractionMessageData[]) => void
 ) => {
-  setExtractionIconData({
-    iconStatus: "warning",
-    iconMessage: "Extraction status pending.",
-  });
   try {
+    setExtractionStatusIcon({
+      iconStatus: undefined,
+      iconMessage: "Loading...",
+    });
     const accessToken = await getAccessToken();
-    const extractions = await extractionClient.getExtractionHistory(accessToken, iModelId, 1);
-    if (extractions.length === 0) {
-      setExtractionIconData({
+    const extraction = extractionClient.getExtractionHistoryIterator(accessToken, iModelId, 1);
+    if((await extraction.next()).done){
```

Comment:
Do `extraction.next()` only once and assign the result to a `const`. Now you should be getting an unwanted result - the second to last extraction jobId should be set on line 55. The first `next` reads the latest extraction jobId. The second next will read the second and so on.

Reply:
Oh, I just realized that logic. I just assigned `extraction.next()` to `latestExtractionResult`.